openid-4-verifiable-credential-issuance-1_0.md

```diff
@@ -597,6 +619,221 @@ Location: https://client.example.net/cb?
+
+#### Require Presentation {#iar-require-presentation}
+
+If `type` is set to `openid4vp_presentation`, as shown in the following example, the response MUST further include an `openid4vp_presentation` parameter containing an Authorization Request as defined in Section 5 of [@!OpenID4VP], encoded in JSON.
+
+```
+HTTP/1.1 200 OK
+Content-Type: application/json
+Cache-Control: no-store
+
+{
+  "status": "require_interaction",
+  "type": "openid4vp_presentation",
+  "auth_session": "123456789",
+  "openid4vp_presentation": {
+    "client_id": "x509_san_dns:rp.example.com",
+    "request_uri": "https://rp.example.com/oidc/request/1234"
+    "response_type": "vp_token",
+    "response_mode": "direct_post",
+    "response_uri": "https//client.example.org/cb",
+    "dcql_query": {
+      "credentials": [
+        {
+          "id": "some_identity_credential",
+          "format": "dc+sd-jwt",
+          "meta": {
+            "vct_values": [ "https://credentials.example.com/identity_credential" ]
+          },
+          "claims": [
+              {"path": ["last_name"]},
+              {"path": ["first_name"]}
+          ]
+        }
+      ]
+    },
+    "nonce": "n-0S6_WzA2Mj"
+  }
+}
+```
+
+The Wallet MUST process the Authorization Request contained in the `openid4vp_presentation` parameter as defined in [@!OpenID4VP] to perform a Credential Presentation to the Authorization Server.
+
+For the requested Presentation, the Issuer is acting as a Verifier to the Wallet.
+The exact architecture and the deployment of the Issuer's OpenID4VP Verifier is out of scope of this specification.
+
+In the Authorization Request contained in the `openid4vp_presentation` parameter, the `response_type` MUST be `vp_token` (defined in Section 8 of [@!OpenID4VP]).
+
+The `response_mode` of the request MUST be `direct_post` (defined in Section 8.2 of [@!OpenID4VP]) or `direct_post.jwt` (defined in Section 8.3.1 of [@!OpenID4VP]).
+Following the definition of this response mode, the Wallet sends an HTTP POST request to the endpoint indicated by the `response_uri` parameter in the Authorization Request to either complete the presentation (sending the VP Token) or to indicate an error.
+In either case, the Verifier responds by sending a JSON object containing a `redirect_uri` parameter.
+
+In a regular presentation flow, the Wallet would be expected to follow this redirect.
+In the case described here, the Wallet MUST NOT follow the redirect URI and MUST instead repeat the request to the Interactive Authorization Endpoint and in this request include the received redirect URI in the `openid4vp_redirect_uri` parameter.
+The Issuer MUST verify that the redirect URI is correct, i.e., matches the one sent in response to the request to the `response_uri`.
+Since the redirect URI MUST include a fresh, cryptographically random value, this check helps to present Session Fixation attacks, see (#iar-security).
```

Comment:
I wonder if more guidance along the following lines would add clarity:

If the Authorization Response to the Verifier was successful, then the Interactive Authorization Endpoint should be an Authorization Code Response. Otherwise, the  Interactive Authorization Endpoint should be an error or an alternative Interaction Required Response